### PR TITLE
support corgi MAVEN component download_url

### PR DESCRIPTION
**Pull request by RichardXuan**
Fixes OLCS-636

### CODE REVIEW

File: openlcsd/flow/tasks.py [hunk 1]
```diff
@@ -366,6 +368,44 @@ def download_shared_remote_source(context, engine):
...
+    component = context.get('component')
+    download_url = component.get('download_url')
+    if not download_url:
+        raise RuntimeError(f"component {component['purl']} download_url empty")
+    name_list = list_http_files(download_url)
+    pom_filename, jar_filename, sources_jar_filename = None, None, None
+    for name in name_list:
+        if name.endswith(".pom"):
+            pom_filename = name
+        if name.endswith(".jar") and "doc" not in name:
```
Comment (qduanmu) [inline openlcsd/flow/tasks.py:382]: `The 'doc' not in name` excludes all jar files with 'doc' in its name including xdocx.jar, is this expected? could you add a comment for this line please?

File: openlcsd/flow/tasks.py [hunk 2]
```diff
@@ -366,6 +368,44 @@ def download_shared_remote_source(context, engine):
...
+            jar_filename = name
+        if name.endswith("sources.jar"):
+            sources_jar_filename = name
```
Comment (qduanmu) [inline openlcsd/flow/tasks.py:385]: Does this mean there would be no more than 1 pom/jar/sources.jar found in the download_url of maven component?

Comment (RichardXuan) [reply openlcsd/flow/tasks.py:385]: I am not sure if there some component have multiple pom/jar/sources.jar. Do u have some example.

File: openlcsd/flow/tasks.py (hunk 1 above)
Comment (RichardXuan) [reply openlcsd/flow/tasks.py:382]: That's not excepted. Thanks for your review, i will change the code

**Comment by qduanmu** [comment]
LGTM

**Pull request merged**